storage/store.go

```diff
@@ -1252,3 +1262,18 @@ func raftEntryFormatter(data []byte) string {
 	}
 	return s
 }
+
+// updateStoreStatus updates the store's status proto.
+func (s *Store) updateStoreStatus() error {
+	now := s.clock.Now().WallTime
+	s.status.UpdatedAt = now
+	s.status.RaftIDs = []int64{}
+	for raftID := range s.ranges {
+		s.status.RaftIDs = append(s.status.RaftIDs, raftID)
+	}
+	key := engine.StoreStatusKey(int32(s.Ident.StoreID))
+	if err := engine.MVCCPutProto(s.engine, nil, key, proto.ZeroTimestamp, nil, s.status); err != nil {
```

Comment:
You need to write the store status using the store.db key value client. Remember that the key value client can write to the distributed database. Writing directly to the engine, as here, may work in unittests but won't work more generally.